nmos-test.py

```diff
@@ -450,9 +450,18 @@ def format_test_results(results, format):
                      "timestamp": time.time(),
                      "results": []}
         for test_result in results["result"]:
-            formatted["results"].append({"name": test_result.name,
-                                         "state": str(test_result.state),
-                                         "detail": test_result.detail})
+            if test_result is None:
+                print(
+                    "The following results currently are being returned: {}"
+                    .format([result.name for result in results["result"] if result != test_result])
+                )
+                raise AttributeError("None object returned as result from one of the tests")
+            else:
+                formatted["results"].append({
+                    "name": test_result.name,
+                    "state": str(test_result.state),
+                    "detail": test_result.detail
+                })

```

Comment:
I was tired of not knowing which test wasn't returning a proper result and causing a crash so this just prints out a list of all the results that ARE returning and used that to work it out

Reply:
Doesn't this just get executed if you're using JSON output? Is it not relevant to the others?